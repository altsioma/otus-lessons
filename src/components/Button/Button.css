.button {
  border-radius: 4px;
  padding: 5px 10px;
  margin: 0 5px;
  background-color: white;
  color: black;
  cursor: pointer;
  border: none;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: box-shadow 0.3s cubic-bezier(.25,.8,.25,1);
};

.button:hover {
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}


.button.button--active {
  position: relative;
  top: 1px;
  background-color: #f0f0f0;
  box-shadow: 0 -1px 0px rgba(0,0,0,0.12), 0 1px 1px rgba(0,0,0,0.24);
}