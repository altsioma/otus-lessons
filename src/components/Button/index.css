.button {
  border: 1px solid black;
  padding: 5px 10px;
  color: white;
  background-color: #c3c3c3;
}