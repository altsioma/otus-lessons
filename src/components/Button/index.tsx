import React from "react";
// import "./index.css";

type ButtonType = {
  textButton: string;
  onClick: () => void;
};

export class Button extends React.Component<ButtonType> {
  constructor(props: ButtonType) {
    super(props);
  }

  render() {
    return (
      <button className="button" onClick={this.props.onClick}>
        {this.props.textButton}
      </button>
    );
  }
}
