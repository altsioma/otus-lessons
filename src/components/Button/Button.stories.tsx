import React from "react";
import { action } from "@storybook/addon-actions";
import { withKnobs, text, boolean, optionsKnob } from "@storybook/addon-knobs";
import { Button } from "./Button";
import { ColorSetType } from "../../types";

export default {
  title: "GraphMakerComponents",
  decorators: [withKnobs],
};

const label = "Colors";
const options: Record<ColorSetType, string> = {
  red: "red",
  green: "green",
  blue: "blue",
  violet: "violet",
};
const defaultValue = "red";

const value = optionsKnob(
  label,
  options,
  defaultValue,
  {
    display: "inline-radio",
  },
  "Colors"
);

export const GraphButton: React.FC<{}> = () => (
  <Button
    color={value as ColorSetType}
    textButton={value}
    isActive={boolean("Active", false)}
    onClick={action("Button click")}
  />
);
