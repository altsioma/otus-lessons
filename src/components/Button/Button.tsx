import React from "react";
import { ButtonStyleType, ButtonStyled } from "./Button.styled";

type ButtonType = ButtonStyleType & {
  textButton: string;
  onClick?: () => void;
};

export const Button: React.FC<ButtonType> = ({ textButton, ...props }) => {
  return <ButtonStyled {...props}>{textButton}</ButtonStyled>;
};
