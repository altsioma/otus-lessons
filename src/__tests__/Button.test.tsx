import React from "react";
import { mount } from "enzyme";
import { Button } from "../components/Button/Button";

const onClick = jest.fn();
const wrapper = mount(
  <Button color="red" textButton="red" onClick={onClick}></Button>
);
describe("Button render check", () => {
  it("Default red button render", () => {
    expect(wrapper.html()).toMatchInlineSnapshot(
      '<button class="button" style="color: red;">red</button>',
      '<button class="button" style="color: red;">red</button>'
    );
  });
  it("Button on click event work", () => {
    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
  it("Active red button has class button--active", () => {
    const wrapper = mount(
      <Button
        color="red"
        textButton="red"
        onClick={onClick}
        isActive={true}
      ></Button>
    );
    expect(wrapper.find(".button").hasClass("button--active")).toBe(true);
  });

  it("Button with color parameter 'green' has color 'green' ", () => {
    const wrapper = mount(
      <Button
        color="green"
        textButton="green"
        onClick={onClick}
        isActive={true}
      ></Button>
    );
    expect(wrapper.props()).toHaveProperty("color", "green");
  });

  it("Button with textButton parameter 'violet' has text 'violet' ", () => {
    const wrapper = mount(
      <Button
        color="violet"
        textButton="violet"
        onClick={onClick}
        isActive={true}
      ></Button>
    );
    expect(wrapper.text()).toEqual("violet");
  });
});
