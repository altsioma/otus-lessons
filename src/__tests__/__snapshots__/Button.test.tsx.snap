// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render check Default red button render: <button class="button" style="color: red;">red</button> 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "button",
      "style": "color: red;",
    },
    "children": Array [
      Object {
        "data": "red",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
